docs/en/operations/settings/settings.md

```diff
@@ -1571,15 +1571,15 @@ Default value: 16.
 
 ## parallel_distributed_insert_select {#parallel_distributed_insert_select}
 
-Enables parallel distributed `INSERT ... SELECT` query. 
+Enables parallel distributed `INSERT ... SELECT` query.
 
 If we execute `INSERT INTO distributed_table_a SELECT ... FROM distributed_table_b` queries and both tables use the same cluster, and both tables are either [replicated](../../engines/table-engines/mergetree-family/replication.md) or non-replicated, then this query is processed locally on every shard.
 
-
 Possible values:
 
 -   0 — Disabled.
--   1 — Enabled.
+-   1 — `SELECT` will be executed on each shard from underlying table of the distributed engine.
+-   2 — `SELECT` and `INSERT` will be executed on each shard from/to underlying table of the distributed engine.
```

Comment:
What do you think about making this setting an enum, i.e., using descriptive names instead of values? I recall we had `SettingEnum` for that. Not sure it will seamlessly support the old constants of 0 and 1, but maybe with some modifications?

Reply:
Yes, the problem with enum is:
- it will not support old value (0, but maybe we can add some quirk for this, or even rename the setting but I want to keep the setting name to make it backward compatible)
- enum value namings :)

Reply:
I think it would be OK to support numbers that correspond to the enum values. This is in `SettingsField.h`, in `SettingField##NEW_NAME##Traits::fromString(const std::string_view & str)`.

Reply:
idk about the names either, disabled/local_source/local_source_and_destination ?

Reply:
>I think it would be OK to support numbers that correspond to the enum values

But then type of the setting will be changed (from int to str), and if [some client](https://github.com/mymarilyn/clickhouse-driver/blob/7c6cd1559ad3a190516581382849fcce652f905d/clickhouse_driver/settings/available.py#L361) uses tcp protocol the setting will not be accepted

Reply:
You're right, I forgot about it. Let's leave this PR as is. If the underlying type of enum settings were Int, we could convert Bool to Enum without breaking the protocol. This is needed sometimes, and having numbers instead of descriptive names is bad for usability. Hopefully we'll be able to think of some support for this conversion in the future.

Reply:
By the way, if we convert int to bool, the native protocol should be fine, but the constants for bool such `'false'` stop working, right? This might be a problem if someone uses it in the profile configuration. Should we support these constants for Int64, for ease of migration?

Reply:
> but the constants for bool such 'false' stop working, right?
>This might be a problem if someone uses it in the profile configuration. Should we support these constants for Int64, for ease of migration?

This is done on the client side AFAICS, am I missing something?

Reply:
The setting is also parsed by the server, if you add it to `users.xml` to some profile. So if the profile has `true` and you change the type to UInt64, the new server will fail to start with the old config. Not sure how much of a nuisance this is, maybe we should just ignore it.

Reply:
You are right, `stringToNumber` can be modified to parse bool always, but not sure that it worth it indeed.